### PR TITLE
BUG: fix compatibility with numpy 2 in diff reports (io.fits, utils)

### PR DESCRIPTION
**Pull request by neutrinoceros**
### Description
This pull request is to address part of #16423 (specifically, failing tests from io.fits)

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Comment by pllim** [comment]
Should this be backported?

**Comment by neutrinoceros** [comment]
Yes

**Review by pllim — APPROVED** [review]
`utils` LGTM. @saimn should double check `io.fits`. Thanks!

File: astropy/io/fits/diff.py [hunk 1]
```diff
@@ -1112,7 +1112,8 @@ def _report(self):
             return
 
         for index, values in self.diff_pixels:
-            index = [x + 1 for x in reversed(index)]
+            # Convert to int to avoid np.int64 in list repr.
+            index = [int(x + 1) for x in reversed(index)]
```
Comment (neutrinoceros) [inline astropy/io/fits/diff.py:1116]: this bit is cherry-picked from https://github.com/astropy/astropy/pull/15065

**Comment by neutrinoceros** [comment]
@pllim I went back to this and replaced my patches with simpler solutions cherry-picked from #15065

File: astropy/io/fits/diff.py (hunk 1 above)
Comment (pllim) [reply astropy/io/fits/diff.py:1116]: In that case, I pinged mhvk for review. Thanks!

**Review by mhvk — APPROVED** [review]
Looks good!

**Comment by mhvk** [comment]
Ah, I see it is smart enough to recognize that the first commit is already in main and is not a problem. So, let's merge this one too.

**Pull request merged**